kythe/docs/kythe-kzip.txt

```diff
@@ -127,3 +127,47 @@ but if set must contain the canonical JSON encoding of an `Index` message.
 For the `proto` encoding of compilation units, the content of the unit
 is the standard wire-encoding of the `kythe.proto.IndexedCompilation`
 protobuf message.
+
+=== Computing the digest for a Compilation Unit
+
+The representative implementation of this is in
+`kythe/go/platform/kcd/kythe/units.go`.
+
+Definitions:
+   `NULL`: the one-byte value corresponding to ASCII NULL (`\0x00`)
+   `NL`:   the one-byte value corresponding to ASCII newline (`\x0a`)
+
+Any strings are emitted as the as described by a UTF-8 encoding.
+
+Let `v` be a `kythe.proto.VName`. Then a digest of `v` is
+
+    `v.signature NULL v.corpus NULL v.root NULL v.path NULL v.language NULL`
+
+Let `cu` be an instance of `kythe.proto.CompilationUnit`. Then a
+digest of `cu` is computed as:
+
+    `"CU" NL cu.vname NULL`
+
+followed by, for each required input:
```

Comment:
Is there a specified sort order for required inputs?

Reply:
required_input is sorted and deduped by (info.digest, info.path)